class.phpmailer.php

```diff
@@ -840,6 +840,13 @@ protected function addAnAddress($kind, $address, $name = '')
             }
             return false;
         }
+        if ($this->hasMultiBytes($address) && function_exists('idn_to_ascii')) {
+            // can't use $this->parseAddresses, because it calls validateAddress, which will fail here
+            list($U,$D) = explode('@',$address);
```

Comment:
That's good. I'd put `function_exists` first as it's probably a cheaper call than `hasMultiBytes`.

Exploding on `@` may run into other issues - it's horrible, but valid, to have an `@` within the local part if it's quoted, like `"a@b"@example.com`. One way around that is to reverse the address before exploding (i.e. split on the rightmost `@`) and limit the explode to 2 parts, then reverse the components.